pkg/oc/cli/cmd/rollout/latest.go

```diff
@@ -42,65 +42,80 @@ var (
-	typer  runtime.ObjectTyper
-	infos  []*resource.Info
+	genericclioptions.IOStreams
+	PrintFlags *genericclioptions.PrintFlags
+
+	Builder   *resource.Builder
+	Namespace string
+	mapper    meta.RESTMapper
+	Resource  string
 
 	DryRun bool
-	out    io.Writer
-	output string
 	again  bool
 
 	appsClient      appsclientinternal.DeploymentConfigsGetter
 	kc              kclientset.Interface
 	baseCommandName string
 
-	// print an object using a printer for a given mapping
-	printObj func(runtime.Object, *meta.RESTMapping, io.Writer) error
+	PrintRevision bool
+	Printer       printers.ResourcePrinter
 }
 
-// NewCmdRolloutLatest implements the oc rollout latest subcommand.
-func NewCmdRolloutLatest(fullName string, f kcmdutil.Factory, out io.Writer) *cobra.Command {
-	opts := &RolloutLatestOptions{
-		baseCommandName: fullName,
+func NewRolloutLatestOptions(name string, streams genericclioptions.IOStreams) *RolloutLatestOptions {
+	return &RolloutLatestOptions{
+		IOStreams:  streams,
+		PrintFlags: genericclioptions.NewPrintFlags("rolled out"),
+
+		baseCommandName: name,
 	}
+}
+
+// NewCmdRolloutLatest implements the oc rollout latest subcommand.
+func NewCmdRolloutLatest(fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRolloutLatestOptions(fullName, streams)
 
 	cmd := &cobra.Command{
 		Use:     "latest DEPLOYMENTCONFIG",
 		Short:   "Start a new rollout for a deployment config with the latest state from its triggers",
 		Long:    rolloutLatestLong,
 		Example: fmt.Sprintf(rolloutLatestExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
-			err := opts.Complete(f, cmd, args, out)
-			kcmdutil.CheckErr(err)
-
-			if err := opts.Validate(); err != nil {
-				kcmdutil.CheckErr(kcmdutil.UsageErrorf(cmd, err.Error()))
-			}
-
-			err = opts.RunRolloutLatest()
-			kcmdutil.CheckErr(err)
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.RunRolloutLatest())
 		},
 	}
 
-	kcmdutil.AddPrinterFlags(cmd)
+	o.PrintFlags.AddFlags(cmd)
+
 	kcmdutil.AddDryRunFlag(cmd)
 	cmd.Flags().Bool("again", false, "If true, deploy the current pod template without updating state from triggers")
 
 	return cmd
 }
 
-func (o *RolloutLatestOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+func (o *RolloutLatestOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
 	if len(args) != 1 {
 		return errors.New("one deployment config name is needed as argument.")
 	}
 
-	namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+	o.Resource = args[0]
+
+	var err error
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
 		return err
 	}
 
 	o.DryRun = kcmdutil.GetFlagBool(cmd, "dry-run")
+	if o.DryRun {
+		o.PrintFlags.Complete("%s (dry run)")
+	}
+	o.Printer, err = o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintRevision = *o.PrintFlags.OutputFormat == "revision"
```

Comment:
@deads2k another snowflake, can we break backwards compatibility here? Is it used at all? If so use a separate flag for it.